metadata/namespaces.go

```diff
@@ -157,32 +176,39 @@ func (s *namespaceStore) Delete(ctx context.Context, namespace string, opts ...n
 	return nil
 }
 
-func (s *namespaceStore) namespaceEmpty(ctx context.Context, namespace string) (bool, error) {
+func (s *namespaceStore) listNamespace(namespace string) (map[string]struct{}, error) {
+	type pair struct {
+		kind   string
+		bucket *bolt.Bucket
+	}
+
 	// Get all data buckets
-	buckets := []*bolt.Bucket{
-		getImagesBucket(s.tx, namespace),
-		getBlobsBucket(s.tx, namespace),
-		getContainersBucket(s.tx, namespace),
+	buckets := []pair{
+		{"images", getImagesBucket(s.tx, namespace)},
+		{"blobs", getBlobsBucket(s.tx, namespace)},
+		{"containers", getContainersBucket(s.tx, namespace)},
 	}
 	if snbkt := getSnapshottersBucket(s.tx, namespace); snbkt != nil {
 		if err := snbkt.ForEach(func(k, v []byte) error {
 			if v == nil {
-				buckets = append(buckets, snbkt.Bucket(k))
+				buckets = append(buckets, pair{"snapshots", snbkt.Bucket(k)})
 			}
 			return nil
 		}); err != nil {
-			return false, err
+			return nil, err
 		}
 	}
 
+	result := make(map[string]struct{})
+
 	// Ensure data buckets are empty
-	for _, bkt := range buckets {
-		if !isBucketEmpty(bkt) {
-			return false, nil
+	for _, pair := range buckets {
+		if !isBucketEmpty(pair.bucket) {
+			result[pair.kind] = struct{}{}
 		}
 	}
 
-	return true, nil
+	return result, nil
```

Comment:
How do you feel about this variation:
```go
func (s *namespaceStore) listNs(namespace string) ([]string, error) {
	var out []string
	if !isBucketEmpty(getImagesBucket(s.tx, namespace)) {
		out = append(out, "images")
	}

	if !isBucketEmpty(getBlobsBucket(s.tx, namespace)) {
		out = append(out, "blobs")
	}

	if snbkt := getSnapshottersBucket(s.tx, namespace); snbkt != nil {
		if err := snbkt.ForEach(func(k, v []byte) error {
			if v == nil {
				if !isBucketEmpty(snbkt.Bucket(k)) {
					out = append(out, "snapshots-"+string(k))
				}
			}
			return nil
		}); err != nil {
			return nil, err
		}
	}

	return out, nil
}
```

- if we want to be informative, let's `k` to clarify which snapshotter.
- no need to remap keys map -> key slice
- no need to have `pair`

Reply:
Looks good. Updated in the last revision.